### PR TITLE
Add Node workflow

### PR DESCRIPTION
**Pull request by Maithem**
A multi-step workflow that adds a new node to the cluster.

### CODE REVIEW

**Comment by Maithem** [comment]
This PR has two dependencies #970 and #977, will rebase after they go in.

File: infrastructure/src/main/java/org/corfudb/infrastructure/orchestrator/AddNodeWorkflow.java [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+     */
+    class AddNodeToLayout implements Action {
+        @Override
+        public String getName() {
+            return "AddNodeToLayout";
+        }
+
+        @Override
+        public Map<String, Object> execute(@Nonnull CorfuRuntime runtime) throws Exception {
+            Layout currentLayout = runtime.getLayoutView().getLayout();
```
Comment (zalokhan) [inline infrastructure/src/main/java/org/corfudb/infrastructure/orchestrator/AddNodeWorkflow.java]: You will need to clone this layout before you pass it to the addNode method.
The add node method tries to seal this layout which will modify it.

Comment (Maithem) [reply infrastructure/src/main/java/org/corfudb/infrastructure/orchestrator/AddNodeWorkflow.java]: There's a race between the invalidate layout and clone.

Comment (zalokhan) [reply infrastructure/src/main/java/org/corfudb/infrastructure/orchestrator/AddNodeWorkflow.java]: Actually `runtime.getLayoutView().getLayout()` should block on the future and wait for the runtime to return the layout after the `invalidateLayout`.
But you are right, there is some race here, I have seen it too but have not been able to spot the exact reason.

**Comment by Maithem** [comment]
subsumed by https://github.com/CorfuDB/CorfuDB/pull/1014

**Pull request closed without merging**